src/common/htmlFileReader.ts

```diff
@@ -11,15 +11,20 @@ export default class HtmlFileReader {
      * Reads the file and returns the string value contained
      * @param file HTML file to read
      */
-    public static readAsText(file: File): Promise<string | ArrayBuffer> {
+    public static readAsText(file: File): Promise<any | ArrayBuffer | File> {
```

Comment:
Doesn't look like the return type matches up any more after this change.  I'd prefer to see an interface mapped to your new `fileInfo` object.

Reply:
So you want me to create an IFileInfo interface in applicationState for this return type? That may also be useful later on when I am loading the project information to my service. I have "any" right now, but wanted to change it to something more specific.